### PR TITLE
fixes #2963 pgsqlutil: schema align fails when there are functions to…

### LINKED ISSUE #2963: pgsqlutil: schema align fails when there are functions to be droped
it accindentally calls getDropSql with a bool inetad of opts hash

```
*** drop trigger trig_workflow_instance from workflow_instance
ST02 /usr/share/qore-modules/0.9.0/PgsqlSqlUtil.qm:1936: RUNTIME-OVERLOAD-ERROR: no variant matching 'PgsqlFunction::getDropSql(bool) can be found; the following variants were tested:
   PgsqlFunction::getDropSql(*hash opt)
   AbstractFunctionBase::getDropSql(*hash opt)
unhandled QORE System exception thrown in TID 1 at 2018-08-29 09:59:57.752788 Wed +02:00 (CEST) in PgsqlTable::getAlignSqlImpl() (/usr/share/qore-modules/0.9.0/PgsqlSqlUtil.qm:1936, user code)
RUNTIME-OVERLOAD-ERROR: no variant matching 'PgsqlFunction::getDropSql(bool) can be found; the following variants were tested:
```

### PR DESCRIPTION
**Pull request by pvanek**
… be droped

### CODE REVIEW

File: examples/test/qlib/SqlUtil/PgsqlSqlUtil.qtest [hunk 1]
```diff
@@ -206,6 +236,7 @@ class PgsqlTest inherits SqlTestBase {
             addTestCase("sequence operators", \testSequenceOperators());
             addTestCase("analytic/window functions", \testAnalyticFunctions());
             addTestCase("default value", \testDefaultValue());
+            addTestCase("pgsqlutil: schema align fails when there are functions to be droped #2963", \testAlignSchemaWithDropFunction());
```
Comment (sejvlond) [inline examples/test/qlib/SqlUtil/PgsqlSqlUtil.qtest]: 120 chars

File: qlib/PgsqlSqlUtil.qm [hunk 2]
```diff
@@ -1868,7 +1868,7 @@ auto v = c.name;
                         l += AbstractDatabase::doCallback(opt, triggers{f.trigger}.getDropSql(name), AbstractDatabase::AC_Drop, "trigger", f.trigger, name);
                         triggers.take(f.trigger);
                     }
-                    l += AbstractDatabase::doCallback(opt, f.getDropSql(opt.force), AbstractDatabase::AC_Drop, "trigger function", f.name, name);
+                    l += AbstractDatabase::doCallback(opt, f.getDropSql(opt), AbstractDatabase::AC_Drop, "trigger function", f.name, name);
```
Comment (sejvlond) [inline qlib/PgsqlSqlUtil.qm:1871]: 120 chars

File: examples/test/qlib/SqlUtil/PgsqlSqlUtil.qtest [hunk 3]
```diff
@@ -89,6 +89,36 @@ class PgsqlTestSchema inherits SqlUtilTestSchema {
...
+# A minimal schema just to create simple table to test bug #2963
+#  = to drop triggers and functions during the schema alignment
+class Bug2963Schema inherits SqlUtilTestSchema {
+    private {
+        const T1 = (
+            "columns": (
+                "id": c_number(14, True, "an example comment with 'quotes'"),
+            ),
+            "primary_key": ("name": "pgsql_test_2963_pk", "columns": ("id")),
+            );
```
Comment (sejvlond) [inline examples/test/qlib/SqlUtil/PgsqlSqlUtil.qtest]: use `{` for hashes

File: examples/test/qlib/SqlUtil/PgsqlSqlUtil.qtest [hunk 4]
```diff
@@ -89,6 +89,36 @@ class PgsqlTestSchema inherits SqlUtilTestSchema {
...
+
+        const TABLES = (
+            "pgsql_test_2963": T1,
+            );
```
Comment (sejvlond) [inline examples/test/qlib/SqlUtil/PgsqlSqlUtil.qtest]: `{` hash

**Review by sejvlond — CHANGES_REQUESTED** [review]

File: examples/test/qlib/SqlUtil/PgsqlSqlUtil.qtest (hunk 3 above)
Comment (pvanek) [reply examples/test/qlib/SqlUtil/PgsqlSqlUtil.qtest]: fixed

File: examples/test/qlib/SqlUtil/PgsqlSqlUtil.qtest (hunk 4 above)
Comment (pvanek) [reply examples/test/qlib/SqlUtil/PgsqlSqlUtil.qtest]: fixed

File: examples/test/qlib/SqlUtil/PgsqlSqlUtil.qtest (hunk 1 above)
Comment (pvanek) [reply examples/test/qlib/SqlUtil/PgsqlSqlUtil.qtest]: fixed

File: qlib/PgsqlSqlUtil.qm (hunk 2 above)
Comment (pvanek) [reply qlib/PgsqlSqlUtil.qm:1871]: wontfix as discussed. It's minimal functional change, no whitespacing to be introduced

**Review by sejvlond — APPROVED** [review]

**Pull request merged**